packages/kokkos-kernels/sparse/src/KokkosSparse_spmv.hpp

```diff
@@ -161,6 +161,12 @@ void spmv(KokkosKernels::Experimental::Controls controls, const char mode[],
     useFallback = useFallback || (mode[0] == Conjugate[0]);
 #endif
   }
+  // cuSPARSE 12 requires that the output (y) vector is 16-byte aligned for all
+  // scalar types
+#if defined(CUSPARSE_VER_MAJOR) && (CUSPARSE_VER_MAJOR == 12)
```

Comment:
Should this be a `>=` comparison? Or do you expect that Cuda will revert this behavior at some point?

Reply:
Since the need for alignment isn't documented in cuSPARSE 12, and this function has the same interface as cuSPARSE 11 (which doesn't seem to need the same alignment), I think there's a good chance that they'll do one of these things in a future version:
- go back to allowing ``sizeof(Scalar)`` alignment
- start documenting the need for 16-byte alignment

In the first case, we don't want to limit performance on future versions by falling back when we don't need to. And in the second, it would be easy to look this up and expand this condition to add version 13.

Reply:
Ok, I agree with you. Maybe adding the test that @maartenarnst uses to Kokkos Kernels' test suite would be a nice addition to this PR, especially if it help you detect any behavioral change with CUDA 13 for instance :smile:

Reply:
Hi @romintomasetti , I just got an update from NVIDIA on this - starting in the next minor version (12.3), spmv will not require the 16-byte alignment, so we can go back to the old behavior. I made a note to narrow the version range for the workaround but it won't matter until that cusparse release (I'm leaving on vacation today)